// @oraichain/common-contracts-sdk/build/Cw1Whitelist.client.js
"use strict";
/**
* This file was automatically generated by @oraichain/ts-codegen@0.35.8.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @oraichain/ts-codegen generate command to regenerate this file.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cw1WhitelistClient = exports.Cw1WhitelistQueryClient = void 0;
class Cw1WhitelistQueryClient {
    constructor(client, contractAddress) {
        this.adminList = async () => {
            return this.client.queryContractSmart(this.contractAddress, {
                admin_list: {}
            });
        };
        this.canExecute = async ({ msg, sender }) => {
            return this.client.queryContractSmart(this.contractAddress, {
                can_execute: {
                    msg,
                    sender
                }
            });
        };
        this.client = client;
        this.contractAddress = contractAddress;
        this.adminList = this.adminList.bind(this);
        this.canExecute = this.canExecute.bind(this);
    }
}
exports.Cw1WhitelistQueryClient = Cw1WhitelistQueryClient;
class Cw1WhitelistClient extends Cw1WhitelistQueryClient {
    constructor(client, sender, contractAddress) {
        super(client, contractAddress);
        this.execute = async ({ msgs }, _fee = "auto", _memo, _funds) => {
            return await this.client.execute(this.sender, this.contractAddress, {
                execute: {
                    msgs
                }
            }, _fee, _memo, _funds);
        };
        this.freeze = async (_fee = "auto", _memo, _funds) => {
            return await this.client.execute(this.sender, this.contractAddress, {
                freeze: {}
            }, _fee, _memo, _funds);
        };
        this.updateAdmins = async ({ admins }, _fee = "auto", _memo, _funds) => {
            return await this.client.execute(this.sender, this.contractAddress, {
                update_admins: {
                    admins
                }
            }, _fee, _memo, _funds);
        };
        this.client = client;
        this.sender = sender;
        this.contractAddress = contractAddress;
        this.execute = this.execute.bind(this);
        this.freeze = this.freeze.bind(this);
        this.updateAdmins = this.updateAdmins.bind(this);
    }
}
exports.Cw1WhitelistClient = Cw1WhitelistClient;
