// @oraichain/common-contracts-sdk/build/Cw20Ics20.client.js
"use strict";
/**
* This file was automatically generated by @oraichain/ts-codegen@0.35.8.
* DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
* and run the @oraichain/ts-codegen generate command to regenerate this file.
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.Cw20Ics20Client = exports.Cw20Ics20QueryClient = void 0;
class Cw20Ics20QueryClient {
    constructor(client, contractAddress) {
        this.port = async () => {
            return this.client.queryContractSmart(this.contractAddress, {
                port: {}
            });
        };
        this.listChannels = async () => {
            return this.client.queryContractSmart(this.contractAddress, {
                list_channels: {}
            });
        };
        this.channel = async ({ id }) => {
            return this.client.queryContractSmart(this.contractAddress, {
                channel: {
                    id
                }
            });
        };
        this.config = async () => {
            return this.client.queryContractSmart(this.contractAddress, {
                config: {}
            });
        };
        this.admin = async () => {
            return this.client.queryContractSmart(this.contractAddress, {
                admin: {}
            });
        };
        this.allowed = async ({ contract }) => {
            return this.client.queryContractSmart(this.contractAddress, {
                allowed: {
                    contract
                }
            });
        };
        this.listAllowed = async ({ limit, startAfter }) => {
            return this.client.queryContractSmart(this.contractAddress, {
                list_allowed: {
                    limit,
                    start_after: startAfter
                }
            });
        };
        this.client = client;
        this.contractAddress = contractAddress;
        this.port = this.port.bind(this);
        this.listChannels = this.listChannels.bind(this);
        this.channel = this.channel.bind(this);
        this.config = this.config.bind(this);
        this.admin = this.admin.bind(this);
        this.allowed = this.allowed.bind(this);
        this.listAllowed = this.listAllowed.bind(this);
    }
}
exports.Cw20Ics20QueryClient = Cw20Ics20QueryClient;
class Cw20Ics20Client extends Cw20Ics20QueryClient {
    constructor(client, sender, contractAddress) {
        super(client, contractAddress);
        this.receive = async ({ amount, msg, sender }, _fee = "auto", _memo, _funds) => {
            return await this.client.execute(this.sender, this.contractAddress, {
                receive: {
                    amount,
                    msg,
                    sender
                }
            }, _fee, _memo, _funds);
        };
        this.transfer = async ({ channel, memo, remoteAddress, timeout }, _fee = "auto", _memo, _funds) => {
            return await this.client.execute(this.sender, this.contractAddress, {
                transfer: {
                    channel,
                    memo,
                    remote_address: remoteAddress,
                    timeout
                }
            }, _fee, _memo, _funds);
        };
        this.allow = async ({ contract, gasLimit }, _fee = "auto", _memo, _funds) => {
            return await this.client.execute(this.sender, this.contractAddress, {
                allow: {
                    contract,
                    gas_limit: gasLimit
                }
            }, _fee, _memo, _funds);
        };
        this.updateAdmin = async ({ admin }, _fee = "auto", _memo, _funds) => {
            return await this.client.execute(this.sender, this.contractAddress, {
                update_admin: {
                    admin
                }
            }, _fee, _memo, _funds);
        };
        this.client = client;
        this.sender = sender;
        this.contractAddress = contractAddress;
        this.receive = this.receive.bind(this);
        this.transfer = this.transfer.bind(this);
        this.allow = this.allow.bind(this);
        this.updateAdmin = this.updateAdmin.bind(this);
    }
}
exports.Cw20Ics20Client = Cw20Ics20Client;
